lib/cocoapods/installer.rb

```diff
@@ -196,23 +196,41 @@ def download_dependencies
 
     private
 
-    def create_generator
-      Xcode::PodsProjectGenerator.new(sandbox, aggregate_targets, pod_targets, analysis_result, installation_options, config)
+    def create_generator(generate_multiple_pod_projects = false)
+      if generate_multiple_pod_projects
+        Xcode::MultiPodsProjectGenerator.new(sandbox, aggregate_targets, pod_targets, analysis_result, installation_options, config)
+      else
+        Xcode::SinglePodsProjectGenerator.new(sandbox, aggregate_targets, pod_targets, analysis_result, installation_options, config)
+      end
     end
 
-    # Generate the 'Pods/Pods.xcodeproj' project.
+    # Generates the Xcode project(s) that go inside the `Pods/` directory.
     #
-    def generate_pods_project(generator = create_generator)
+    def generate_pods_project(generator = create_generator(installation_options.generate_multiple_pod_projects))
       UI.section 'Generating Pods project' do
         pod_project_generation_result = generator.generate!
         @target_installation_results = pod_project_generation_result.target_installation_results
         @pods_project = pod_project_generation_result.project
+        @pod_target_subprojects = pod_project_generation_result.projects_by_pod_targets.keys
+        projects_by_pod_targets = pod_project_generation_result.projects_by_pod_targets
         run_podfile_post_install_hooks
-        project_writer = Xcode::PodsProjectWriter.new(sandbox, pods_project,
-                                                      target_installation_results.pod_target_installation_results,
-                                                      installation_options)
-        project_writer.write!
-        generator.share_development_pod_schemes(@pods_project)
+        pods_project_writer = Xcode::PodsProjectWriter.new(sandbox, pods_project,
+                                                           target_installation_results.pod_target_installation_results,
+                                                           installation_options)
+        pods_project_writer.write!
+        projects_by_pod_targets.each do |project, pod_targets|
+          pod_target_project_writer = Xcode::PodsProjectWriter.new(sandbox, project,
+                                                                   target_installation_results.pod_target_installation_results,
+                                                                   installation_options)
+          pod_target_project_writer.write!
+          generator.share_development_pod_schemes(project, development_pod_targets(pod_targets))
+        end
+
+        # Share the remaining pod targets. Generally this will be none for when `generate_multiple_pod_projects` is set to true
+        # and all pod_targets when set to false.
+        remaining_development_pods = development_pod_targets(Set.new(pod_targets).subtract(Set.new(projects_by_pod_targets.values.flatten)))
```

Comment:
this could be `development_pod_targets(pod_targets - projects_by_pod_targets.values.flatten)`